### PR TITLE
✨ Add EnableWatchBookmarks option to cache informers

### PR DESCRIPTION
**Pull request by vincepri**
AllowWatchBookmarks is generally pretty safe to enable as it has been available in Kuberentes for a long while, and the server ignores the flag if it doesn't implement it (per docs).

### CODE REVIEW

File: pkg/cache/internal/informers.go [hunk 1]
```diff
@@ -444,6 +446,9 @@ func (ip *Informers) makeListWatcher(gvk schema.GroupVersionKind, obj runtime.Ob
 			},
 			// Setup the watch function
 			WatchFunc: func(opts metav1.ListOptions) (watch.Interface, error) {
+				opts.Watch = true // Watch needs to be set to true separately
```
Comment (alvaroaleman) [inline pkg/cache/internal/informers.go:449]: Why is this needed now?

**Review by alvaroaleman — COMMENTED** [review]
Maybe add an opt-out option just in case?

File: pkg/cache/internal/informers.go (hunk 1 above)
Comment (vincepri) [reply pkg/cache/internal/informers.go:449]: I just added it because I noticed it was in the other one, not sure if it is actually needed?

**Review by alvaroaleman — APPROVED** [review]

**Review by alvaroaleman — APPROVED** [review]
